4_experiment_comparision/Explore_dataset.md

```diff
@@ -0,0 +1,363 @@
+```
+
+```python
+df_mmo_r = df_mmo.query("EXPERIMENT==1")
+df_mmo_c = df_mmo.query("EXPERIMENT==0")
+
+compare_two_groups(df_mmo_r, df_mmo_c, "MMO2_DIFF", "percentage difference of MMO 2 day")
+compare_two_groups(df_mmo_r, df_mmo_c, "MMO7_DIFF", "percentage difference of MMO 7 day")
+```
+
+# Face measurement comparision
+
+```python
+print_NA_count(df, ['Sum0','Sum2', 'Sum7'])
+```
+
+```python
+df_sum = df.dropna(subset= ['Sum0','Sum2', 'Sum7'])
+df_sum = df_sum.assign(SUM2_DIFF = lambda x: (x.Sum2 - x.Sum0) / x.Sum0)
+df_sum = df_sum.assign(SUM7_DIFF = lambda x: (x.Sum7 - x.Sum0) / x.Sum0)
+```
+
+```python
+plot_violin(df_sum, ['SUM2_DIFF', 'SUM7_DIFF'], "EXPERIMENT", "Compare SUM' difference to before operation in percents")
+```
+
+```python
+s_research_index = df_sum["EXPERIMENT"] == 1
+s_control_index = df_sum["EXPERIMENT"] == 0
+
+compare_two_groups(df_sum[s_research_index], df_sum[s_control_index], "SUM2_DIFF", "percantage difference of SUM measurements 2 day")
+compare_two_groups(df_sum[s_research_index], df_sum[s_control_index], "SUM7_DIFF", "percantage difference of SUM measurements 7 day")
+```
+
+# Medicine taken comparision
+
+```python
+med_compared_columns = ['Med1','Med2', 'Med7']
+```
+
+```python
+print_NA_count(df, med_compared_columns)
+```
+
+```python
+df_meds = df.dropna(subset=med_compared_columns)
+for col in med_compared_columns:
+    df_meds[col] = df_meds[col].str.extract('(\d+)').astype(int)
+
+```
+
+```python
+df_meds
+```
+
+```python
+plot_violin(df_meds, compared_columns, "EXPERIMENT", "Compare Meds' taken")
+```
+
+```python
+df_meds_r_index = df_meds["EXPERIMENT"] == 1
+df_meds_c_index = df_meds["EXPERIMENT"] == 0
+
+compare_two_groups(df_meds[df_meds_r_index], df_meds[df_meds_c_index], "Med1", "Meds taken 1 day")
+compare_two_groups(df_meds[df_meds_r_index], df_meds[df_meds_c_index], "Med2", "Meds taken 2 day")
+compare_two_groups(df_meds[df_meds_r_index], df_meds[df_meds_c_index], "Med2", "Meds taken 7 day")
+```
+
+# Control group - compare by operator
+
+```python
+prepared_control = df_control.dropna(subset=["Operator", "VAS1", "VAS2", "VAS7", "Med1", "Med2", "Med7"])
+for col in med_compared_columns:
+    prepared_control[col] = prepared_control[col].str.extract('(\d+)').astype(int)
+```
+
+```python
+statistics_vas_by_operator = prepared_control.groupby("Operator")[["VAS1", "VAS2", "VAS7"]].sum()
+statistics_meds_by_operator =  prepared_control.groupby("Operator")[[ "Med1", "Med2", "Med7"]].sum()
+```
+
+```python
+statistics_vas_by_operator.plot.bar()
+statistics_meds_by_operator.plot.bar()
+plt.show()
+```
+
+```python
+statistics_sum_by_operator = prepared_control.groupby("Operator")[["Sum0", "Sum2", "Sum2"]].sum()
+statistics_mmo_by_operator =  prepared_control.groupby("Operator")[[ "MMO0", "MMO2", "MMO7"]].sum()
+```
+
+```python
+statistics_sum_by_operator.plot.bar()
+statistics_mmo_by_operator.plot.bar()
+plt.show()
+```
+
+# Cluster people and compare
```

Comment:
note before running any code:
so the idea here is thats (and lets start with this analysis) we have the column `operator` for control group. The history is that we did not have it before, we only have it because i have noticed that there is something different between groups of patients (it was different operators :D). We want to quantify/backwards engineer the thing - idea is to use output columns i.e (`MMO change`, `sum change`, `meds` and `vas` - perhaps all on 2nd day is enough) and cluster them as a way of exploratory data analysis.  

One thing you wouldn't know is the number of clusters and common way of finding it is knee method e.g. https://practicaldatascience.co.uk/machine-learning/how-to-use-knee-point-detection-in-k-means-clustering

for starters do the knee method and visualize the clusters for each `number_of_clusters`. Note: for 2 and 3d you can do that on 'raw data' if you use more columns:, you can try something like `Cluster visualization` in:  
https://towardsdatascience.com/best-practices-for-visualizing-your-cluster-results-20a3baac7426